### PR TITLE
Fix Presto Issue 5322

### PR DESCRIPTION
**Pull request by ebd2**
See prestodb#5322 for a description of the issue.

Commits need to be squashed prior to merge with prestodb/presto or there will be a commit with failing product tests.

### CODE REVIEW

**Comment by ebd2** [comment]
@dain

**Comment by electrum** [comment]
Please add a non-integration test for this.

**Comment by ebd2** [comment]
@electrum: Requested tests added and rebased.

File: presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java [hunk 1]
```diff
@@ -176,23 +176,34 @@ else if (frameInfo.getType() == RANGE) {
...
+
+        if (startType == FOLLOWING && endType == UNBOUNDED_FOLLOWING) {
+            return getStartValue() > positions;
+        }
+
+        if (startType != endType) {
             return false;
         }
 
-        FrameBound.Type type = frameInfo.getStartType();
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/operator/window/WindowPartition.java]: Let's keep this variable `type` to make it clear that at this point, the frames have the same type

**Comment by electrum** [comment]
Merged, thanks!

**Pull request closed without merging**